pkg/querier/querier.go

```diff
@@ -527,3 +532,46 @@ func (q *Querier) validateQueryTimeRange(userID string, from *time.Time, through
 
 	return nil
 }
+
+func (q *Querier) checkTailRequestLimit(ctx context.Context) error {
+	userID, err := user.ExtractOrgID(ctx)
+	if err != nil {
+		return err
+	}
+
+	replicationSet, err := q.ring.GetAll()
+	if err != nil {
+		return err
+	}
+
+	// we want to check count of active tailers with only one of the active ingesters since
+	// all of them would be having same number of active tail connections
+	// Note: In worst-case scenario the ingester that we picked would have joined recently which might not have all the tail requests
```

Comment:
I think we should check all ingesters to avoid this error case. It doesn't seem like a relatively more expensive option (we're already creating tailers on every ingester for tailing requests). It does however keep us from loading additional queriers with tailers when the only ingester checked is new.

While this seems like it'd work itself out via the law of large numbers, consider a client retrying bad tail requests. This would eventually succeed if ingesters are rolling over, overburdening older ingesters.